docs/source/schema/unions-interfaces.md

```diff
@@ -1,48 +1,84 @@
 ---
 title: Unions and interfaces
-description: How to add unions and interfaces to a schema
+description: Polymorphic schema types
 ---
 
-Unions and interfaces are great when you have fields that are in common between two types.
+**Unions** and **interfaces** are polymorphic GraphQL types that enable a schema field to contain one of multiple object types.
+
+> **Tip:** If a field can contain object types that **all share relevant subfields**, use an interface. Otherwise, use a union.
```

Comment:
This is difficult for me to grok.

Primarily because fields don't contain object types, but they can return them.
Secondly, I think expanding on the two scenarios would be valuable. This may be deserving of a complete section with examples rather than just an aside.

Reply:
It seems what I'm asking for is actually written out below. In which case I'd consider removing this tip or rewording it in a way that's a bit more descriptive and/or easier to understand.